### PR TITLE
Core: Remove one comment from FastAppend

### PR DESCRIPTION
**Pull request by gaborkaszab**
There is a comment for FastAppend that mentions the number of commit retries done by the class. I removed it for 2 reasons:
  1) Not the FastAppend class does the commit so the commit retries shouldn't be commented there.
  2) The number of commit retries is configurable via a table property so mentioning a hard-coded
     value in the comment is incorrect.

Added a new test that adjusts the COMMIT_NUM_RETRIES table property. Adjusted existing tests to exercise the boundaries of the number of commit retries.

### CODE REVIEW

File: core/src/test/java/org/apache/iceberg/TestFastAppend.java [hunk 1]
```diff
@@ -252,11 +252,36 @@ public void testFailure() {
...
+    validateSnapshot(null, readMetadata().currentSnapshot(), FILE_B);
+  }
+
   @TestTemplate
   public void testAppendManifestCleanup() throws IOException {
-    // inject 5 failures
+    // inject failures
     TestTables.TestTableOperations ops = table.ops();
-    ops.failCommits(5);
+    ops.failCommits(TableProperties.COMMIT_NUM_RETRIES_DEFAULT + 1);
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/TestFastAppend.java:284]: TBH I don't see much value in updating this and the other tests

File: core/src/main/java/org/apache/iceberg/FastAppend.java [hunk 2]
```diff
@@ -24,20 +24,14 @@
...
 import org.apache.iceberg.io.InputFile;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.Iterables;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 import org.apache.iceberg.util.CharSequenceSet;
 
-/**
- * {@link AppendFiles Append} implementation that adds a new manifest file for the write.
- *
- * <p>This implementation will attempt to commit 5 times before throwing {@link
```
Comment (nastra) [inline core/src/main/java/org/apache/iceberg/FastAppend.java:38]: why not rather update the comment and refer to the table property that is being used for retries: `TableProperties#COMMIT_NUM_RETRIES_DEFAULT`? 
In your PR description you mention that retries aren't done for `FastAppend`, but retries are done for all subclasses of `SnapshotProducer`.
Also if we update this javadoc, then we should align this for other subclasses of `SnapshotProducer` as well

File: core/src/test/java/org/apache/iceberg/TestFastAppend.java (hunk 1 above)
Comment (gaborkaszab) [reply core/src/test/java/org/apache/iceberg/TestFastAppend.java:284]: The value here is that we can test boundary values for the config and not just use a value that is within the range.

File: core/src/main/java/org/apache/iceberg/FastAppend.java (hunk 2 above)
Comment (gaborkaszab) [reply core/src/main/java/org/apache/iceberg/FastAppend.java:38]: Thanks for taking a look, @nastra!
Frankly, I don' really prefer redundant comments. I don't see why we'd want to add the same comment for all the subclasses of SnapshotProducer. First, as said in my description retries don't happen on that level of abstraction. Let's say for some reason the retry logic or the name of the config changes in the future. Then we have to remember that there is a comment about this in all the subclasses that we have to change now too.

What if I add a comment about the retries to SnapshotProducer itself? That's the level of abstraction where that logic happens, would be more suitable and we wouldn't have redundant comments in the subclasses.

**Comment by gaborkaszab** [comment]
Hi @nastra ,
Do you disagree with my reasoning?

File: core/src/test/java/org/apache/iceberg/TestFastAppend.java (hunk 1 above)
Comment (gaborkaszab) [reply core/src/test/java/org/apache/iceberg/TestFastAppend.java]: Makes sense. I reverted the changes in existing tests.
Thanks for taking a look!

**Review by nastra — APPROVED** [review]

**Comment by gaborkaszab** [comment]
Thanks for taking a look, @nastra !
